src/components/Widgets/Markdown/serializers/index.js

```diff
@@ -164,10 +165,30 @@ export const remarkToMarkdown = obj => {
    */
   const mdast = obj || u('root', [u('paragraph', [u('text', '')])]);
 
+  const remarkToMarkdownPluginOpts = {
+    commonmark: true,
+    fences: true,
+    pedantic: true,
+    listItemIndent: '1',
+
+    // Settings to emulate the defaults from the Prosemirror editor, not
+    // necessarily optimal. Should eventually be configurable.
+    bullet: '*',
+    strong: '*',
+    rule: '-',
+  };
+
+  /**
+   * Escape markdown entities found in text and html nodes within the MDAST.
+   */
+  const escapedMdast = unified()
+    .use(remarkEscapeMarkdownEntities)
+    .runSync(mdast);
```

Comment:
I don't understand how this actually works, just because I'm not familiar with ReMark, but it looks fine to me.

Reply:
Yeah, basically remark/unified lets you run parsers via `.parse()`, plugins via `.run()/.runSync()`, stringifiers via `.stringify()`, or all three together via `.process()/.processSync()`. Because of our use case, we never run all three together.

What I've done here is applied a plugin (synchronous, no need for async here) to the MDAST (MarkDown AST). The plugin is passed the entire tree, and it recursively transforms the contents of text nodes by escaping markdown entities.